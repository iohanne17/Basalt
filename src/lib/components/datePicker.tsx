import React, { Fragment, useState } from 'react'
import { Pressable, StyleSheet } from 'react-native'
import { Text } from '@Components/index'
import { Theme } from '@Theme/theme'
import Ionicons from 'react-native-vector-icons/Ionicons'
import { colors } from '@Theme/colors'
import DatePicker from 'react-native-date-picker'

export interface DatePickerProps {
  title: string
  onConfirm: (val: Date) => void
}

export const DatePickerComponent = ({ title, onConfirm }: DatePickerProps) => {
  const [date, setDate] = useState(new Date())
  const [openDate, setOpen] = useState(false)

  const DatePickerButton = ({
    title,
    onPress,
    date = null,
  }: {
    title: string
    onPress: () => void
    date: string | null
  }) => {
    return (
      <Pressable onPress={onPress} style={s.picker}>
        <Text text_semibold>{title}</Text>
        <Ionicons name={'calendar'} size={Theme.sizes.icon3} color={colors.primary} />
        <Text text_semibold>{date || ''}</Text>
      </Pressable>
    )
  }

  return (
    <Fragment>
      <DatePickerButton title={title} onPress={() => setOpen(true)} date={date.toDateString()} />
      <DatePicker
        modal
        open={openDate}
        date={date}
        onConfirm={date => {
          setOpen(false)
          setDate(date)
          onConfirm(date)
        }}
        onCancel={() => {
          setOpen(false)
        }}
      />
    </Fragment>
  )
}

export default DatePickerComponent

const s = StyleSheet.create({
  picker: {
    paddingHorizontal: Theme.sizes.h1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    alignSelf: 'center',
  },
})
