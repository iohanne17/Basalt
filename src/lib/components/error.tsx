import * as React from 'react';
import {Theme} from '../theme';
import {View, StyleSheet} from 'react-native';
import {Text} from './Text/Text';
import {Spacer} from './Spacer';
import {Button} from './Button/MainButton';

interface ErrorProps {
  message: string;
  title: string;
  isLoading?: boolean;
  onPress?: () => void;
  showButton?: boolean;
}

export const ErrorScreen = ({
  message,
  title,
  isLoading,
  onPress,
  showButton = true,
}: ErrorProps) => (
  <View style={styles.loadingContainer}>
    <Text title>{title}</Text>
    <Text>{message}</Text>
    <Spacer height={20} />
    {showButton ? (
      <Button
        loading={isLoading}
        onPress={() => onPress?.()}
        type="text"
        title="Retry"
      />
    ) : null}
  </View>
);

const styles = StyleSheet.create({
  loadingContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Theme.colors.light.inverseWhite100,
  },
});

export default ErrorScreen;
