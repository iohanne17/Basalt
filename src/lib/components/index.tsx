export * from './loading';
export * from './Button/MainButton';
export * from './Layout';
export * from './Text/Text';
export * from './Spacer';
export * from './error';
export * from './network';
export * from './ListItemSeparator/ListItemSeparator';
export * from './row';
