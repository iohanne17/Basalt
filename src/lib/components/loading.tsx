import * as React from 'react';
import {Theme} from '../theme';
import {View, StyleSheet, ActivityIndicator} from 'react-native';

export const Loading = () => (
  <View style={styles.loadingContainer}>
    <ActivityIndicator
      color={Theme.colors.light.primary}
      size={'large'}
      animating={true}
    />
  </View>
);

const styles = StyleSheet.create({
  loadingContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Theme.colors.light.inverseWhite100,
  },
});

export default Loading;
