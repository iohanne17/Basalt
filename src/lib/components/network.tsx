import * as React from 'react'
import { Theme } from '../theme'
import { View, StyleSheet } from 'react-native'
import { Text } from './Text/Text'

export const NetErrorScreen = () => (
  <View style={styles.loadingContainer}>
    <Text title>{'Network Error'}</Text>
    <Text>{'Unable to connect... Try again later'}</Text>
  </View>
)

const styles = StyleSheet.create({
  loadingContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Theme.colors.light.inverseWhite100,
  },
})

export default NetErrorScreen
