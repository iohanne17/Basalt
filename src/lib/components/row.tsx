import * as React from 'react';
import {Theme} from '../theme';
import {Text} from './Text/Text';
import {View, StyleSheet} from 'react-native';

interface RowProps {
  title: string;
  value: string | number;
}

export const Row = ({title, value}: RowProps) => {
  return (
    <View style={s.row}>
      <Text text style={s.rowTitle}>
        {title}
      </Text>
      <Text text_semibold style={s.rowValue}>
        {value}
      </Text>
    </View>
  );
};

const s = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: Theme.sizes.v2,
  },
  rowTitle: {
    textTransform: 'capitalize',
  },
  rowValue: {
    textTransform: 'uppercase',
    color: Theme.colors.dark.inverseWhite100,
  },
});

export default Row;
