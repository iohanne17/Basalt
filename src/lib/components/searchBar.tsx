import { Theme } from '@Theme/theme'
import { useDebounce } from '@Utils/userUtils'
import React, { FC, useEffect, useState } from 'react'
import { View, TextInput, StyleSheet, TextInputProps, TouchableOpacity } from 'react-native'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons'

interface SearchBarProps extends TextInputProps {
  onSearch: (query: string) => void
  debounceTime?: number
  onCancel?: () => void
}

export const SearchBar: FC<SearchBarProps> = ({ onSearch, debounceTime = 800, onCancel, ...textInputProps }) => {
  const [query, setQuery] = useState<string | undefined>(undefined)
  const searchInput = useDebounce(query, 1000)

  useEffect(() => {
    if (searchInput && searchInput.length > 0) {
      onSearch(searchInput)
    }
  }, [searchInput])

  const onPressCancel = () => {
    handleTextChange?.('')
    onCancel?.()
  }

  const handleTextChange = (text: string) => {
    setQuery(text)
    if (text === '') {
      onCancel?.()
    }
  }

  return (
    <View style={styles.container}>
      <TextInput
        placeholder="Search stock by name or ticker symbol..."
        value={query}
        style={styles.input}
        onChangeText={val => {
          setQuery(val)
          handleTextChange(val)
        }}
        {...textInputProps}
      />
      {query && query?.length > 0 && (
        <TouchableOpacity onPress={onPressCancel}>
          <MaterialIcon name="cancel" color={Theme.colors.light.primary} size={Theme.sizes.icon3} />
        </TouchableOpacity>
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    padding: 10,
    backgroundColor: Theme.colors.light.searchBar,
    justifyContent: 'center',
    flexDirection: 'row',
    alignItems: 'center',
  },
  input: {
    height: 40,
    letterSpacing: 1.5,
    fontWeight: '500',
    fontFamily: 'SchibstedGrotesk-Bold',
    flex: 1,
  },
})

export default SearchBar
