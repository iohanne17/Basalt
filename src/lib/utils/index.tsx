export * from './navigationUtils';
export * from './userUtils';
