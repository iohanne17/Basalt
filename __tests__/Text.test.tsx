import React from 'react';
import {render} from '@testing-library/react-native';
import {Text} from '../src/lib/components';

enum TextColor {
  main = 'colorMain',
  primary = 'primary',
  error = 'error',
  black = 'black',
  white = 'white',
  fade = 'fade',
}

describe('Text component', () => {
  it('renders children', () => {
    const {getByText} = render(<Text>Hello, world!</Text>);
    const textElement = getByText('Hello, world!');
    expect(textElement).toBeTruthy();
  });

  it('matches snapshot', () => {
    const {toJSON} = render(
      <Text color={TextColor.primary}>Hello, snapshot!</Text>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
