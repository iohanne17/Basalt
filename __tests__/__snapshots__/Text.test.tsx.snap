// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component matches snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#FE812A",
      "fontFamily": "SchibstedGrotesk-Medium",
      "fontSize": 17,
      "letterSpacing": -0.22,
      "lineHeight": 24,
    }
  }
>
  Hello, snapshot!
</Text>
`;
